src/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -634,6 +634,32 @@ public void TestStartInfo()
             }
         }
 
+        [Theory]
+        [InlineData(@"""abc"" d e", @"abc,d,e")]
+        [InlineData(@"""abc""      d e", @"abc,d,e")]
+        [InlineData("\"abc\"\t\td\te", @"abc,d,e")]
+        [InlineData(@"a\\b d""e f""g h", @"a\\b,de fg,h")]
+        [InlineData(@"\ \\ \\\", @"\,\\,\\\")]
+        [InlineData(@"a\\\""b c d", @"a\""b,c,d")]
+        [InlineData(@"a\\\\""b c"" d e", @"a\\b c,d,e")]
+        [InlineData(@"a""b c""d e""f g""h i""j k""l", @"ab cd,ef gh,ij kl")]
+        [InlineData(@"a b c""def", @"a,b,cdef")]
```

Comment:
One particular edge case I ran into while looking into this was quotes in the middle of backslashes which act as arg delimiters (not as literal quotes). Looking at everything I don't think that scenario will break, but it may be worth adding as a test.

An example:
`[InlineData(@"""\a\"" \\""\\\ b", @"\a"" \\\\,b")]`